### ISSUE TITLE
Fading model fails for `P2G4_MOD_BLE2M` modulation

### ISSUE THREAD

**Issue by rstoica** [BabbleSim/ext_2G4_channel_Indoorv1#1]
Thank you very much for this work and enabling realistic indoor 2G4 channel models within babblesim.

#### Problem
I have encountered an issue related to modeling fading  for the BLE 2M PHY, i.e. for `P2G4_MOD_BLE2M`. The error is:

```p_2G4: ERROR: (src/channel_2G4Indoorv1.c:380): Unknown modulation type 32```

#### Reproducing
An easy way to reproduce it is running
```
./bs_central_hr_device_nrf52 -d=0 -s=trial &
./bs_peripheral_device_nrf52 -d=1 -s=trial &
./bs_2G4_phy_v1 -s=trial -D=2 -channel=Indoorv1 -sim_length=10e6 &
```
with the nrf52_bsim builds of the zephyr central_hr and peripheral bluetooth samples.

#### Fix
The issue is in the switch statement at https://github.com/BabbleSim/ext_2G4_channel_Indoorv1/blob/master/src/channel_2G4Indoorv1.c#L348 where the case for `P2G4_MOD_BLE2M` is not considered, so a quick fix is adding the case statement:
```
    case P2G4_MOD_BLE2M:
      BW = 2*samplesPerMHz; break;
```

**Comment by kelsayed** [comment]
Thanks rstoica for this fix. I had the same issue and applied your fix and it works perfectly. 
I think you should file a pull request.

**Comment by aescolar** [comment]
Thanks @rstoica & @kelsayed . I completely missed that you had created these issues.
Indeed proper 2Mbps BLE support had never been added here. These models predate the BT SIG releasing 5.0, and at that time the support for what was going to be 2Mbps BLE in the simulator was thru the "proprietary" 2Mbps modulation.
It is now fixed adafc2c566ca34beb7aa9e9b6352f5b598594b87.

Closing the issue as fixed.
